l10n_it_fatturapa_pec/models/sdi.py

```diff
@@ -70,20 +78,19 @@ def check_pec_server_id(self):
     def check_email_validity(self):
         if self.env.context.get("skip_check_email_validity"):
             return
-        for channel in self:
+        pec_channels = self.filtered(lambda c: c.channel_type == 'pec')
+        for channel in pec_channels:
             if not extract_rfc2822_addresses(channel.email_exchange_system):
                 raise exceptions.ValidationError(
                     _("Email %s is not valid") % channel.email_exchange_system
                 )
 
     def check_first_pec_sending(self):
-        sdi_address = self.env["ir.config_parameter"].get_param("sdi.pec.first.address")
         if not self.first_invoice_sent:
-            if self.email_exchange_system != sdi_address:
-                raise exceptions.UserError(
-                    _("This is a first sending but SDI address is different " "from %s")
-                    % sdi_address
```

Comment:
@SirTakobi questo check lo rimuovi del tutto?

Reply:
Sì, ho rimosso il controllo perché si basava sul default del campo `email_exchange_system`, ma il default assegna quel valore a ogni canale SdI indipendentemente dal tipo, mentre è un valore che interessa solo ai canali `pec`.

Ho rimosso quindi il controllo `if self.email_exchange_system != sdi_address -> raise(...)` ma, in base alla stessa condizione, poco sotto assegno `self.email_exchange_system = sdi_address`.